relay_messages.go

```diff
@@ -226,7 +247,7 @@ func (f *lazyCallReq) Arg2Iterator() (arg2.KeyValIterator, error) {
 		return arg2.KeyValIterator{}, fmt.Errorf("non thrift scheme %s", f.as)
 	}
 
-	if f.arg2EndOffset > int(f.Header.PayloadSize()) {
+	if f.arg2EndOffset > f.Header.PayloadSize() {
```

Comment:
might be worth adding a TODO here, but I don't think this can happen anymore since we now have the rbuf + check for error right? we might be able to remove this error check.